build.gradle

```diff
@@ -23,6 +22,30 @@ plugins {
 	id 'java' // for jd gen
 }
 
+def toolchainLanguageVersion = JavaLanguageVersion.of(11)
+
+if (!JavaVersion.current().isCompatibleWith(JavaVersion.toVersion(toolchainLanguageVersion))) {
+	logger.warn ":The current Java Version ${JavaVersion.current()} is too old."
+	logger.warn ":Gradle may download newer Java online and the build may be slower."
+}
+
+java.toolchain { // defined in JavaPluginExtension
+	languageVersion = toolchainLanguageVersion
+}
+
+// currently gradle does not yet propagate that version above, so
+tasks.withType(JavaExec) {
+	javaLauncher = javaToolchains.launcherFor {
+		languageVersion = toolchainLanguageVersion
+	}
+}
+
+tasks.withType(Javadoc) {
+	javadocTool = javaToolchains.javadocToolFor {
+		languageVersion = toolchainLanguageVersion
+	}
+}
```

Comment:
This does use java 11 when using 15 for example, might be worth moving this to only apply when running on java versions bellow 11 (same time as the warning).

Reply:
I will just set the current java language version as the toolchain language version if it can compile or run the minimum version (this case 11). Done in last forced update.